components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOImpl.java

```diff
@@ -0,0 +1,320 @@
+                                    webhook.getEventSchemaName());
+                            statement.setString(WebhookSQLConstants.Column.EVENT_SCHEMA_URI,
+                                    webhook.getEventSchemaUri());
+                            statement.setString(WebhookSQLConstants.Column.EVENT_SCHEMA_VERSION,
+                                    WEBHOOK_SCHEMA_VERSION);
+                            statement.setString(WebhookSQLConstants.Column.STATUS, String.valueOf(webhook.getStatus()));
+                            statement.setInt(WebhookSQLConstants.Column.TENANT_ID, tenantId);
+                        }, webhook, true));
+    }
+
+    private void updateWebhookInDB(Webhook webhook, int tenantId) throws TransactionException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        jdbcTemplate.withTransaction(template -> {
+            template.executeUpdate(WebhookSQLConstants.Query.UPDATE_WEBHOOK,
+                    statement -> {
+                        statement.setString(WebhookSQLConstants.Column.ENDPOINT, webhook.getEndpoint());
+                        statement.setString(WebhookSQLConstants.Column.DESCRIPTION, webhook.getDescription());
+                        statement.setString(WebhookSQLConstants.Column.SECRET, webhook.getSecret());
+                        statement.setString(WebhookSQLConstants.Column.VERSION, WEBHOOK_VERSION);
+                        statement.setString(WebhookSQLConstants.Column.EVENT_SCHEMA_NAME, webhook.getEventSchemaName());
+                        statement.setString(WebhookSQLConstants.Column.EVENT_SCHEMA_URI, webhook.getEventSchemaUri());
+                        statement.setString(WebhookSQLConstants.Column.EVENT_SCHEMA_VERSION, WEBHOOK_SCHEMA_VERSION);
+                        statement.setString(WebhookSQLConstants.Column.STATUS, String.valueOf(webhook.getStatus()));
+                        statement.setString(WebhookSQLConstants.Column.UUID, webhook.getUuid());
+                        statement.setInt(WebhookSQLConstants.Column.TENANT_ID, tenantId);
+                    });
+            return null;
+        });
+    }
+
+    private int getInternalWebhookIdByUuid(String uuid, int tenantId) throws TransactionException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        return jdbcTemplate.withTransaction(template ->
+                template.fetchSingleRecord(WebhookSQLConstants.Query.GET_WEBHOOK_INTERNAL_ID_BY_ID,
+                        (resultSet, rowNumber) -> resultSet.getInt(WebhookSQLConstants.Column.ID),
+                        statement -> {
+                            statement.setString(WebhookSQLConstants.Column.UUID, uuid);
+                            statement.setInt(WebhookSQLConstants.Column.TENANT_ID, tenantId);
+                        }));
+    }
```

Comment:
I don't think this method is needed

Reply:
Regarding [1], getInternalWebhookIdByUuid is necessary to add events for a specific webhook ID, as the foreign key pertains to the ID. The same pattern is followed in the rule-mgt component [2] as well.

[1] https://github.com/wso2/carbon-identity-framework/pull/6741#discussion_r2100053903
[2] https://github.com/wso2/carbon-identity-framework/blob/master/components/rule-mgt/org.wso2.carbon.identity.rule.management/src/main/java/org/wso2/carbon/identity/rule/management/internal/dao/impl/RuleManagementDAOImpl.java#L242